### PR TITLE
Add Dockerfiles, Kubernetes YAML and Travis file

### LINKED ISSUE #2: Add Dockerfile
Add Dockerfile with multi-stage build and Go 1.10.4

### LINKED ISSUE #3: Add Kubernetes YAML file
Add Kubernetes YAML file

### PR DESCRIPTION
**Pull request by ivanayov**
This adds Dockerfiles, build scripts, Travis build and connector
deployment for Kubernetes

Signed-off-by: Ivana Yovcheva <iyovcheva@vmware.com>

Fixes #3 
Fixes #2

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+COPY vendor     vendor
+COPY main.go    .
+
+# Run a gofmt and exclude all vendored code.
+RUN test -z "$(gofmt -l $(find . -type f -name '*.go' -not -path "./vendor/*"))"
+
+RUN go test -v ./...
+
+# Stripping via -ldflags "-s -w" 
+RUN CGO_ENABLED=0 GOOS=linux go build -a -ldflags "-s -w" -installsuffix cgo -o ./connector
```
Comment (alexellis) [inline Dockerfile]: I think we should switch to `alpine:3.8 to` ship it. Could you try that out?

File: yaml/connector-dep.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: vcenter
+        component: vcenter-connector
+    spec:
+      containers:
+      - name: vcenter
+        image: functions/vcenter-connector:latest
```
Comment (alexellis) [inline yaml/connector-dep.yml]: Could we start with the 0.1.0 tag?

File: yaml/connector-dep.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        env:
+          - name: vcenter-url
```
Comment (alexellis) [inline yaml/connector-dep.yml]: This needs to be an "arg" - see the openfaas-operator or the of-builder / of-buildkit YAML files as an example of how to pass args.

File: yaml/connector-dep.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            value: http://user:pass@127.0.0.1:8989/sdk
+          - name: insecure
```
Comment (alexellis) [inline yaml/connector-dep.yml]: This needs to be an arg too

File: Dockerfile (hunk 1 above)
Comment (alexellis) [reply Dockerfile]: To clarify I mean use multi-sage build like the gateway.

Comment (alexellis) [reply Dockerfile]: I don't believe so

**Comment by ivanayov** [comment]
@alexellis it's now updated

**Comment by alexellis** [comment]
Please can you use the instructions for running this with the Docker image you generated to verify it works correctly?

https://github.com/openfaas-incubator/vcenter-connector#example

**Comment by alexellis** [comment]
This is only partially working, but I'm going to merge. @embano1 has a working Docker image for vcsim for Kubernetes. Let's find out what he did differently.

**Pull request merged**